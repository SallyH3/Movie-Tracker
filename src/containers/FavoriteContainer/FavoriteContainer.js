import React from 'react';
import { connect } from 'react-redux';
import Movie from '../../components/Movie/Movie';

const FavoriteContainer = (props) => {
  const image = props.movies.map(movie => {
    if(movie.favorited) {
      return <Movie {...movie}/>
    }
  })
  return(
    <section className='main'>
      <section className='image-container'>
        {image}
      </section>
    </section>
  )
}

const mapStateToProps = (state) => ({
  movies: state.movies
})

export default connect(mapStateToProps)(FavoriteContainer)



