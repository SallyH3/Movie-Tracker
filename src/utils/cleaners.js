export const cleanMovies = (movies) => {
  return movies.map(movie => {
    return {
      movie_id: movie.id,
      release_date: movie.release_date,
      vote_average: movie.vote_average,
      title: movie.title,
      poster_path: 'https://image.tmdb.org/t/p/w300_and_h450_bestv2/' + movie.poster_path,
      overview: movie.overview,
      favorited: false,
      key: movie.id
    };
  });
}