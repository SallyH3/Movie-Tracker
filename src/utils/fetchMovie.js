export const fetchMovie = (url) => {
  return fetch(url)
  .then(response => {
    if(!response.ok) {
      throw Error('Error fetching movie')
    } else {
      return response.json();
    }
  })
}