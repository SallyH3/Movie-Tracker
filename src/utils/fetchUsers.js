export const fetchUsers = (url, options) => {
  return fetch(url, options)
  .then(response => {
    if(!response.ok) {
      return 'error'
    } else {
      return response.json()
    }
  })
}

//comment again