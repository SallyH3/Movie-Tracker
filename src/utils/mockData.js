export const mockUser = {
	id: 1,
	name: 'Taylor',
	password: 'password',
	email: "tman2272@aol.com"
}

