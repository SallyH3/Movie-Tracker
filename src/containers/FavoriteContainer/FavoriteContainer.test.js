import React from 'react';
import { FavoriteContainer } from './FavoriteContainer';
import { shallow } from 'enzyme';
import { mapStateToProps } from './FavoriteContainer';
import { mockSelectedMovie } from '../../utils/mockData';

describe('FavoriteContainer', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(
      <FavoriteContainer
        movies={ [mockSelectedMovie] }/>
    )
    expect(wrapper).toMatchSnapshot();
  });
  describe('mapStateToProps', () => {
    it('should return a props object with movies', () => {

      const mockState= {
        movies: 'movies'
      }

      const expected = {
        movies: 'movies'
      }

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected)
    })
  })
})