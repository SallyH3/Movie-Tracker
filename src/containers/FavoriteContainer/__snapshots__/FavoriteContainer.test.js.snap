// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteContainer should match snapshot 1`] = `ShallowWrapper {}`;
